sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -632,9 +632,51 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     val rawTable = getRawTable(db, table)
     val withNewSchema = rawTable.copy(schema = schema)
     verifyColumnNames(withNewSchema)
+
+    val newProperties = collection.mutable.Map[String, String]()
+    for ((property, value) <- withNewSchema.properties) {
+      newProperties.put(property, value)
+    }
+    for ((property, value) <- tableMetaToTableProps(withNewSchema)) {
+      newProperties.put(property, value)
+    }
+
+    // Since the table schema will be changed, check if the bucketing and sort columns are still
```

Comment:
Unfortunately YES. Below I alter a bucketed table to remove the bucketing column from its schema. Modified table still points to old bucket spec. This is bad.

```
hive> desc formatted bucketed_partitioned_1;

# col_name            	data_type           	comment
user_id             	bigint
name                	string

# Storage Information
....
Num Buckets:        	8
Bucket Columns:     	[user_id]
Sort Columns:       	[Order(col:user_id, order:1)]

hive> ALTER TABLE bucketed_partitioned_1 REPLACE COLUMNS ( eid INT, name STRING);
OK

hive> desc formatted bucketed_partitioned_1;
# col_name            	data_type           	comment
eid                 	int
name                	string

# Storage Information
.....
Num Buckets:        	8
Bucket Columns:     	[user_id]
Sort Columns:       	[Order(col:user_id, order:1)]
```